public/app/features/datasources/components/picker/DataSourcePicker.tsx

```diff
@@ -78,6 +80,16 @@ export function DataSourcePicker(props: DataSourcePickerProps) {
   const [filterTerm, setFilterTerm] = useState<string>('');
   const { onKeyDown, keyboardEvents } = useKeyNavigationListener();
   const ref = useRef<HTMLDivElement>(null);
+  const debouncedTrackSearch = useMemo(
+    () =>
+      debounce((q) => {
+        reportInteraction(INTERACTION_EVENT_NAME, {
+          item: INTERACTION_ITEM.SEARCH,
+          query: q,
+        });
+      }, 300),
```

Comment:
300 seems to ideal for typing https://stackoverflow.com/questions/42361485/how-long-should-you-debounce-text-input